mm2src/coins/eth/v2_activation.rs

```diff
@@ -33,6 +36,23 @@ impl From<AbortedError> for EthActivationV2Error {
     fn from(e: AbortedError) -> Self { EthActivationV2Error::InternalError(e.to_string()) }
 }
 
+impl From<CryptoCtxError> for EthActivationV2Error {
+    fn from(e: CryptoCtxError) -> Self { EthActivationV2Error::InternalError(e.to_string()) }
+}
+
+/// An alternative to `crate::PrivKeyActivationPolicy`, typical only for ETH coin.
+#[derive(Clone, Deserialize)]
+pub enum EthPrivKeyActivationPolicy {
+    ContextPrivKey,
+    Metamask,
```

Comment:
Should this be compiled conditionally with `#[cfg(target_arch = "wasm32")]`?

Reply:
I think it's worth to return an error if `Metamask` policy is used natively:
https://github.com/KomodoPlatform/atomicDEX-API/blob/cb4aabadd95ad6fa9d3cd01385f1f96b5558387c/mm2src/coins_activation/src/eth_with_token_activation.rs#L265-L269

Reply:
The `unknown variant` will be returned on native platforms. We can also mention in the docs that `Metamask` variant exists only on WASM.

Reply:
Done